src/components/menu/menu.tsx

```diff
@@ -67,6 +105,37 @@ export class Menu {
     @Prop({ reflect: true })
     public gridLayout = false;
 
+    /**
+     * A search function that takes a search-string as an argument,
+     * and returns a promise that will eventually be resolved with
+     * an array of `MenuItem`:s.
+     *
+     * See the docs for the type `Searcher` for type information on
+     * the searcher function itself.
+     */
+    @Prop()
+    public searcher: MenuSearcher;
+
+    /**
+     * Making it possible for a consumer to tell the component that it's waiting for something.
+     * For example loading items/root items.
+     */
+    @Prop()
+    public loading: boolean;
```

Comment:
What happened with this? Did you try and change the `item` prop to `Array<MenuItem<T> | ListSeparator> | MenuLoader` to get rid of this prop?

Reply:
yes, we tried it out and it got really complex really fast for the consumer. And also pretty complex in the component itself :/

Reply:
We tried it and it turned really complex here, but mostly for the consumer.